### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by dusty3ntity**
**Issue link**

[#306 Technical debt: refactoring](https://github.com/ita-social-projects/dokazovi-fe/issues/306)

## Summary of issue

- [x] Declare types for query params (`directions`, `regions`, `page`, `types`)
- [x] Remove redundant mapping in the data slice
- [x] Remove optional modifiers in the declared types and remove the corresponding checks
- [x] Better API types naming
- [x] Refactor all the filters components (`PostTypesFilter`, `TagsFilter`, `CheckBoxDropdownFilterForm`, `CheckBoxFilterForm`)
- [x] Fix document title updates
- [x] Fix entity type declarations using `string` as the `id` type instead of `number`
- [x] Fix `setupExpertsMaterialsId` action fires on every `ExpertMaterialsContainer` component rerender

## Summary of change

- [x] Declare types for query params (`directions`, `regions`, `page`, `types`)
- [x] Remove redundant mapping in the data slice
- [x] Remove optional modifiers in the declared types and remove the corresponding checks
- [x] Better API types naming
- [x] Refactor all the filters components (`PostTypesFilter`, `TagsFilter`, `CheckBoxDropdownFilterForm`, `CheckBoxFilterForm`)
- [x] Fix document title updates
- [x] Fix entities using `string` as the `id` type instead of `number`
- [x] Fix `setupExpertsMaterialsId` action fires on every `ExpertMaterialsContainer` component rerender
- [x] Refactor the theme typography and the PostView styles
- [x] Create a wrapper component, same as for `PostView`, for the `ExpertProfileView` to avoid the "not found" error
- [x] Improve the loading indicator components
- [x] Add validation on query params
- [x] Fix uploading an image by url
- [x] Fix initial logging in fires two times
- [x] Fix scrolling to the "load more" button on the expert profile page

### CODE REVIEW

**Review by SergeOlabin — APPROVED** [review]

File: src/lib/components/Filters/CheckboxDropdownFilterForm.tsx [hunk 1]
```diff
@@ -13,23 +13,26 @@ import {
...
   id: string | number;
   name: string;
 }
 
-export interface ICheckBoxDropdownFilterFormProps {
-  onFormChange: (checked: ICheckBoxFormState, filterType?) => void;
+export interface ICheckboxDropdownFilterFormProps {
+  onFormChange: (
+    checked: ICheckboxFormState,
+    filterType: FilterTypeEnum,
```
Comment (SergeOlabin) [inline src/lib/components/Filters/CheckboxDropdownFilterForm.tsx:27]: same as CheckBoxFilterForm

File: src/lib/components/Footer/Footer.tsx [hunk 2]
```diff
@@ -17,26 +17,21 @@ interface IHeaderProps {
 const navElems: IHeaderProps[] = [
   {
     id: '1',
```
Comment (SergeOlabin) [inline src/lib/components/Footer/Footer.tsx:19]: mb numbers?

File: src/lib/components/Loading/LoadingContainer.styles.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    },
+    expandedContainer: {
+      flexDirection: 'column',
+      alignItems: 'center',
+      justifyContent: 'center',
+      flexGrow: 1,
+    },
+  }),
+  { name: 'LoadingContainer' },
+);
```
Comment (SergeOlabin) [inline src/lib/components/Loading/LoadingContainer.styles.ts:18]: if you don't need theming you can just pass config ojb
```suggestion
export const useStyles = makeStyles({
    container: {
      flexDirection: 'column',
      alignItems: 'center',
      justifyContent: 'center',
    },
    expandedContainer: {
      flexDirection: 'column',
      alignItems: 'center',
      justifyContent: 'center',
      flexGrow: 1,
    },
  },
  { name: 'LoadingContainer' },
);
```

**Review by SergeOlabin — APPROVED** [review]

File: src/lib/components/Editor/CustomModules/VideoUrlInputModal.tsx [hunk 4]
```diff
@@ -36,11 +36,8 @@ const VideoUrlInputModal: React.FC<IVideoUrlInputModalProps> = ({
...
   };
 
-  const canParseId = (url) => {
-    if (parseVideoIdFromUrl(url)) {
-      return true;
-    }
-    return false;
+  const canParseId = (url: string) => {
+    return Boolean(parseVideoIdFromUrl(url));
   };
```
[the remark this answers is not in the document]
Comment (dusty3ntity) [reply src/lib/components/Editor/CustomModules/VideoUrlInputModal.tsx:41]: Done

File: src/lib/components/Footer/Footer.tsx (hunk 2 above)
Comment (dusty3ntity) [reply src/lib/components/Footer/Footer.tsx:19]: Of course! Haven't noticed it

File: src/lib/components/Loading/LoadingContainer.styles.ts (hunk 3 above)
Comment (dusty3ntity) [reply src/lib/components/Loading/LoadingContainer.styles.ts:18]: Done

File: src/lib/components/Filters/CheckboxFilterForm.tsx [hunk 5]
```diff
@@ -1,33 +1,39 @@
...
+export interface ICheckboxFormState extends Record<string, boolean> {}
+
+export interface ICheckboxFilterFormProps {
+  onFormChange: (
+    checked: ICheckboxFormState,
+    filterType: FilterTypeEnum,
+  ) => void;
   possibleFilters: IFilter[];
   selectedFilters?: IFilter[];
+  filterType: FilterTypeEnum;
```
[the remark this answers is not in the document]
Comment (dusty3ntity) [reply src/lib/components/Filters/CheckboxFilterForm.tsx:19]: Decoupled the filter components with the `FilterTypeEnum`. Now using it in the components where these filters are used, in order to have only one function to process all the filters' changes. See `MaterialsView`, for example

File: src/lib/components/Filters/CheckboxDropdownFilterForm.tsx (hunk 1 above)
Comment (dusty3ntity) [reply src/lib/components/Filters/CheckboxDropdownFilterForm.tsx:27]: Decoupled the filter components with the `FilterTypeEnum`. Now using it in the components where these filters are used, in order to have only one function to process all the filters' changes. See `MaterialsView`, for example

**Review by AnnaMinasian — APPROVED** [review]

**Pull request merged**